jakarta.ls/src/main/java/io/microshed/jakartals/commons/SnippetRegistry.java

```diff
@@ -0,0 +1,306 @@
+     */
+    private static Snippet createSnippet(
+        JsonReader reader,
+        TypeAdapter<? extends ISnippetContext<?>> contextDeserializer)
+    throws JsonIOException, JsonSyntaxException {
+        GsonBuilder builder = new GsonBuilder();
+        builder.registerTypeAdapter(Snippet.class, new SnippetDeserializer(contextDeserializer));
+        return builder.create().fromJson(reader, Snippet.class);
+    }
+
+    /**
+	 * Returns all snippets.
+	 *
+	 * @return all snippets.
+	 */
+	public List<Snippet> getSnippets() {
+		return snippets;
+	}
+/**
+	 * Returns the snippet completion items according to the context filter.
+	 *
+	 * @param replaceRange       the replace range.
+	 * @param lineDelimiter      the line delimiter.
+	 * @param canSupportMarkdown true if markdown is supported to generate
+	 *                           documentation and false otherwise.
+	 * @param contextFilter      the context filter.
+	 * @return the snippet completion items according to the context filter.
+	 */
+	public List<CompletionItem> getCompletionItem(final Range replaceRange, final String lineDelimiter,
+			boolean canSupportMarkdown, Predicate<ISnippetContext<?>> contextFilter) {
+        // TODO Add context based filtering
+        return getSnippets().stream().map(snippet -> { 
+            String label = snippet.getPrefixes().get(0);
+            CompletionItem item = new CompletionItem();
+            item.setLabel(label);
+            item.setDetail(snippet.getDescription());
+            String insertText = getInsertText(snippet, false, lineDelimiter);
+            item.setKind(CompletionItemKind.Snippet);
+            item.setDocumentation(Either.forRight(createDocumentation(snippet, canSupportMarkdown, lineDelimiter)));
+            item.setFilterText(label);
+			item.setTextEdit(new TextEdit(replaceRange, insertText));
+            item.setInsertTextFormat(InsertTextFormat.Snippet);
+            return item;
+        }).collect(Collectors.toList());
+	}
+
+	public List<CompletionItem> getCompletionItemNoContext(final Range replaceRange, final String lineDelimeter, boolean canSupportMarkdown) {
+		return getSnippets().stream().map(snippet -> {
+			// To filter by context, I just need to provide document contexts, and then perform a match and include or remove
+			// List<String> snippetTypes = ((SnippetContextForJava) snippet.getContext()).getTypes();
+			// LOGGER.info(snippetTypes.toString());
+			String label = snippet.getPrefixes().get(0);
+			CompletionItem item = new CompletionItem();
+            item.setLabel(label);
+            item.setDetail(snippet.getDescription());
+			String insertText = getInsertText(snippet, false, lineDelimeter);
+            item.setKind(CompletionItemKind.Snippet);
+            item.setDocumentation(Either.forRight(createDocumentation(snippet, canSupportMarkdown, lineDelimeter)));
+            item.setFilterText(label);
+			item.setTextEdit(new TextEdit(replaceRange, insertText));
```

Comment:
@AnkushSharma2698  Ryan and I discussed and have come up with this solution to fix the indentation for now.  Seems that formatting the snippets based on where they are inserted is an additional feature that VS Code offers, Eclipse does not have the same support.  

For now, on the server side we want to use the range to determine the indentation.
ie. if the range looks like the following, we would use the character value to add in \t characters to our newText, such that the newText would now be something like: `@GET\n\t@Produces(MediaType.TEXT_PLAIN)\n\tpublic String ${1:methodname}() {\n\t	return "hello";\n\t}`
```
TextEdit [
  range = Range [
    start = Position [
      line = 19
      character = 1
    ]
    end = Position [
      line = 19
      character = 1
    ]
  ]
  newText = "@GET\n@Produces(MediaType.TEXT_PLAIN)\npublic String ${1:methodname}() {\n	return "hello";\n}"
]
```

Reach out with any questions!